Devscord.DiscordFramework/Framework/Commands/Parsing/CommandParser.cs

```diff
@@ -48,26 +50,96 @@ private string GetName(string message)
 
         private IEnumerable<DiscordRequestArgument> GetArguments(string message)
         {
-            var prefix = this.GetPrefix(message);
-            var splitted = message.Split(prefix)
-                .Where(x => !string.IsNullOrWhiteSpace(x))
-                .Select(x => x.Trim());
-            return splitted.Select(x => this.GetArgument(x, prefix));
+            var arguments = new List<DiscordRequestArgument>();
+            var trimmedMess = message.Trim();
+
+            for (var i = 0; i < trimmedMess.Length; i++)
+            {
+                var startsWithArg = _possiblePrefixes.Any(x => trimmedMess[i..].TrimStart().StartsWith(x));
+                if (!startsWithArg)
+                {
+                    var arg = new DiscordRequestArgument
+                    {
+                        Prefix = null,
+                        Name = null,
+                        Values = GetValues(trimmedMess[i..], out var lastValuesIndex)
+                    };
+                    i += lastValuesIndex;
+                    arguments.Add(arg);
+                    continue;
+                }
+
+                foreach (var prefix in _possiblePrefixes)
+                {
```

Comment:
jak masz tak duży foreach to lepiej podzielić to na kilka metod 
albo wrzucić to do osobnej metody, dla czytelności